# Assignment 1

## Overview

  Kickstarter is an American public-benefit corporation based in Brooklyn, New York, that maintains a global crowdfunding platform focused on creativity. The company's stated mission is to "help bring creative projects to life". Kickstarter has reportedly received more than $1.9 billion in pledges from 9.4 million backers to fund 257,000 creative projects, such as films, music, stage shows, comics, journalism, video games, technology and food-related projects. 
  People who back Kickstarter projects are offered tangible rewards or experiences in exchange for their pledges. This model traces its roots to subscription model of arts patronage, where artists would go directly to their audiences to fund their work. We are trying to analyze the kickstarter dataset to extract meaningful plots to illustate the data. This analysis can help us to gradually predict which attributes make a project successful. You can find more information about this dataset from <a link="https://www.kaggle.com/kemical/kickstarter-projects">here</a>.   

In [1]:
    # loading the dataset
    import pandas as pd
    import numpy as np
    df = pd.read_csv('dataset/ks-projects-201612-cleaned.csv',  sep = ',')

ModuleNotFoundError: No module named 'pandas'

In [4]:
print(df.head())

   uniqueId                                               name  \
0      5971              An American Apocalypse: The Reckoning   
1     18520                                 Grandma's are Life   
2     21109                                               Meta   
3     21371                                  Water Powered Car   
4     24380  Puss N' Books: A relaxing cat cafe and bookstore.   

          category main_category currency          deadline     goal  \
0           Comics        Comics      USD    8/6/2012 20:00  48000.0   
1      World Music         Music      USD  11/18/2016 15:32  15000.0   
2  Performance Art           Art      GBP    5/6/2015 23:00    150.0   
3       Technology    Technology      USD   8/10/2016 19:02  75000.0   
4           Spaces          Food      USD  11/26/2015 17:25  20000.0   

           launched  pledged       state  backers country  
0    6/8/2012 18:46    376.0      failed        7      US  
1  10/19/2016 15:32     62.0      failed        4 

In [5]:
print (df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 319510 entries, 0 to 319509
Data columns (total 12 columns):
uniqueId         319510 non-null int64
name             319508 non-null object
category         319510 non-null object
main_category    319510 non-null object
currency         319510 non-null object
deadline         319510 non-null object
goal             319510 non-null float64
launched         319510 non-null object
pledged          319510 non-null float64
state            319510 non-null object
backers          319510 non-null int64
country          319510 non-null object
dtypes: float64(2), int64(2), object(8)
memory usage: 29.3+ MB
None


In [6]:
print (df.dtypes)


uniqueId           int64
name              object
category          object
main_category     object
currency          object
deadline          object
goal             float64
launched          object
pledged          float64
state             object
backers            int64
country           object
dtype: object


### Summary Statistics:

In [7]:
df.describe()

uniqueId          goal       pledged        backers
count  3.195100e+05  3.195100e+05  3.195100e+05  319510.000000
mean   1.074803e+09  4.763327e+04  8.767841e+03     102.760079
std    6.193723e+08  1.145984e+06  9.008617e+04     940.099326
min    5.971000e+03  1.000000e-02  0.000000e+00       0.000000
25%    5.374061e+08  2.000000e+03  3.000000e+01       2.000000
50%    1.075764e+09  5.000000e+03  6.100000e+02      12.000000
75%    1.611034e+09  1.500000e+04  3.939000e+03      55.000000
max    2.147476e+09  1.000000e+08  2.033899e+07  219382.000000

### Exploratory Data Analysis 
 There are 17 columns in our dataset. Following are the columns that we use:

<table> 
<tr>
<td>
<b>column name</b>                         
</td>
<td><b>Description</b><td>
<td><b>Type</b></td>
</tr>
<tr>
<td>uniqueID</td>
<td>Id of project </td>
<td>String</td>
</tr>

<tr>
<td>name</td>
<td>Title of the project</td>
<td>String</td>
</tr>
<tr>
<td>category</td>
<td>Kickstarter sub category</td>
<td>String</td>
</tr>
<tr>
<td>main_category</td>
<td>Kickstarter main category</td>
<td>String</td>
</tr>
<tr>
<td>currency</td>
<td>Intial currency of the project</td>
<td>String</td>
</tr>
<tr>
<td>deadline</td>
<td>Project deadline</td>
<td>DateTime</td>
</tr>
<tr>
<td>goal</td>
<td>Project goal</td>
<td>Numeric</td>
</tr>
<tr>
<td>launched</td>
<td>Project launch date</td>
<td>DateTime</td>
</tr>
<tr>
<td>pledged</td>
<td>Amount pledge in inital currency</td>
<td>Numeric</td>
</tr>
<tr>
<td>state</td>
<td>Project current state</td>
<td>String</td>
</tr>
<tr>
<td>backers</td>
<td>Number of backers</td>
<td>Numeric</td>
</tr>
<tr>
<td>country</td>
<td>Country of project creator</td>
<td>String</td>
</tr>

<tr>
<td>usd pledged</td>
<td>Kickstarter conversion of pledged amount in USD</td>
<td>Numeric</td>
</tr>
</table>






### Status of the project
 There are 319510 projects in this dataset out of which 113,061 have been succesful

In [8]:
a = len(df)
c = len(df[df.state=="canceled"])
f =  len(df[df.state=="failed"])
s = len(df[df.state=="successful"])
print "Out of %s projects, %s are cancelled, %s have failed, and %s have been successful."%(a, c, f, s)

Out of 319510 projects, 32343 are cancelled, 168203 have failed, and 113061 have been successful.
